Practical Example/IosInterviewPracticeProject/IosInterviewPracticeProject/Services/CourseAPIService.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  NetworkManager.swift
+//  IosInterviewPracticeProject
+//
+//  Created by Robert Ramirez on 8/20/20.
+//  Copyright © 2020 me.robert.ramirez. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class CourseAPIService {
+  static let shared = CourseAPIService()
+  static let artworkCache = NSCache<NSString, UIImage>()
+  private let dateFormatter: DateFormatter = {
+    let formatter = DateFormatter()
+    formatter.locale = .current
+    formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+    return formatter
+  }()
+
+  private let decoder: JSONDecoder = {
+    let decoder = JSONDecoder()
+    decoder.keyDecodingStrategy = .convertFromSnakeCase
+    return decoder
+  }()
+
+  func fetchContent(for url: String, completion: @escaping (Result<Results, CourseAPIServiceError>) -> Void) {
+    guard let videosEndPointURL = URL(string: url) else { return }
+
+    URLSession.shared.dataTask(with: videosEndPointURL) { data, _, error in
```

Comment:
Just as a suggestion, think about using the new URLSession Combine API to do fetching and decoding. https://www.raywenderlich.com/7864801-combine-getting-started#toc-anchor-007

Reply:
I have not started learning combine. I will look into it.